google-ads-examples/src/main/java/com/google/ads/googleads/examples/accountmanagement/GetAccountHierarchy.java

```diff
@@ -0,0 +1,360 @@
+
+      // Optional: You may pass the loginCustomerId on the command line or specify a loginCustomerId
+      // here if and only if the managerId is set. If the loginCustomerId is set neither on the
+      // command line nor below, a null value will be passed to the run example method, and the
+      // example will use each respective accessible customer ID as the loginCustomerId.
+      // params.managerId = Long.parseLong("INSERT_MANAGER_ID_HERE");
+    }
+
+    if (params.managerId != null && params.loginCustomerId == null) {
+      System.err.println("loginCustomerId must be provided if managerId is provided.");
+      return;
+    } else if (params.managerId == null && params.loginCustomerId != null) {
+      System.err.println("loginCustomerId may not be provided if managerId is not provided.");
+      return;
+    }
+
+    GoogleAdsClient googleAdsClient;
+    try {
+      googleAdsClient = GoogleAdsClient.newBuilder().fromPropertiesFile().build();
+    } catch (FileNotFoundException fnfe) {
+      System.err.printf(
+          "Failed to load GoogleAdsClient configuration from file. Exception: %s%n", fnfe);
+      return;
+    } catch (IOException ioe) {
+      System.err.printf("Failed to create GoogleAdsClient. Exception: %s%n", ioe);
+      return;
+    }
+
+    try {
+      new GetAccountHierarchy()
+          .runExample(googleAdsClient, params.managerId, params.loginCustomerId);
+    } catch (GoogleAdsException gae) {
+      // GoogleAdsException is the base class for most exceptions thrown by an API request.
+      // Instances of this exception have a message and a GoogleAdsFailure that contains a
+      // collection of GoogleAdsErrors that indicate the underlying causes of the
+      // GoogleAdsException.
+      System.err.printf(
+          "Request ID %s failed due to GoogleAdsException. Underlying errors:%n",
+          gae.getRequestId());
+      int i = 0;
+      for (GoogleAdsError googleAdsError : gae.getGoogleAdsFailure().getErrorsList()) {
+        System.err.printf("  Error %d: %s%n", i++, googleAdsError);
+      }
+    }
+  }
+
+  /**
+   * Creates a new GoogleAdsClient instance with the specified loginCustomerId.
+   *
+   * @param loginCustomerId the loginCustomerId used to create the GoogleAdsClient.
+   * @return a GoogleAdsClient instance.
+   */
+  private GoogleAdsClient createGoogleAdsClient(long loginCustomerId) {
+    GoogleAdsClient googleAdsClient;
+    try {
+      googleAdsClient =
+          GoogleAdsClient.newBuilder()
+              .fromPropertiesFile()
+              .setLoginCustomerId(loginCustomerId)
+              .build();
+    } catch (FileNotFoundException fnfe) {
+      System.err.printf(
+          "Failed to load GoogleAdsClient configuration from file. Exception: %s%n", fnfe);
+      return null;
+    } catch (IOException ioe) {
+      System.err.printf("Failed to create GoogleAdsClient. Exception: %s%n", ioe);
+      return null;
```

Comment:
I think that's a design decision. At the outset, we decided to support partial failure, in a way, because we skip test accounts that don't produce hierarchies. There are additional reasons that a hierarchy may not be fetched (e.g. a client account with an incomplete billing setup). Going off of the initial decision to support partial failure, I've taken the following approach.

First, we get a list of all accessible customers. In the event that the GAC cannot be created, the entire example will fail in the same way most code examples fail. Once we have the list of accessible customers, we loop through each. For each one that we can retrieve an account hierarchy, we'll add it to a list of hierarchies. For each one that fails, for whatever reason, we'll add to a list of failed cases. Then, we'll print all of the IDs of the accounts that failed, followed by the hierarchies of all the IDs that were successful.

This case would be an example of a single ID failing. I don't see why this would be different than the other reasons retrieving the hierarchy might fail, so that is why I've treated it this way.

The other reasons why I chose this design are because it is parallel to how the initial GAC is created in the main method - the method just returns. In this case, we must return null because the return value is a GAC object rather than void. In addition, because we list all accessible accounts at the outset, I like the idea of covering them all in the subsequent print statements. Each customer ID will be found in either the successful cases or the failures.

All of that said, I think the main point here is that we're "supporting partial failure" for some cases, so why should this case be different and fail the entire example?

For now, I am going to leave this as as but add some more comments and also add the CID in the error message. However, let me know what you think.